src/settings/toml/deploy_target.rs

```diff
@@ -0,0 +1,155 @@
+use crate::settings::toml::Route;
+
+#[derive(Debug)]
+pub struct RouteConfig {
+    pub workers_dev: Option<bool>,
+    pub route: Option<String>,
+    pub routes: Option<Vec<String>>,
+    pub zone_id: Option<String>,
+    pub account_id: Option<String>,
+}
+
+impl RouteConfig {
+    fn has_conflicting_targets(&self) -> bool {
+        if self.workers_dev.unwrap_or_default() {
+            self.routes_defined()
+        } else {
+            if let Some(pattern) = &self.route {
+                !pattern.is_empty() && self.routes.is_some()
+            } else {
+                false
+            }
+        }
+    }
+
+    pub fn routes_defined(&self) -> bool {
+        if self.routes.is_some() {
+            true
+        } else if let Some(pattern) = &self.route {
+            !pattern.is_empty()
+        } else {
+            false
+        }
+    }
+
+    pub fn is_zoneless(&self) -> bool {
+        self.workers_dev.unwrap_or_default() && !self.has_conflicting_targets()
+    }
+
+    pub fn is_zoned(&self) -> bool {
```

Comment:
is there a reason the `is_zoneless` method includes a check on `has_conflicting_targets` and `zoned` does not?

Reply:
previously they both did but they should probably not.